### PR TITLE
write definitions for glossary terms

### PR DESCRIPTION
**Pull request by jjelschen**

### CODE REVIEW

File: te_markdown/terminology.md [hunk 1]
```diff
@@ -35,25 +35,47 @@ In this area, details about the test step selected in the _Test Execution Naviga
...
-* Tester
-
-<!--- TODO: write user-friendly definitions, layout the glossary as definition list with anchors for cross-referencing. The following is an example of how that might look:
-<a name="domain-expert"></a>Domain expert
-: A stakeholder and probably a user of the system under test, who has a deep understanding of the problem domain. Domain experts specify what they expect the system to do and how it should behave, using the [test specification language (TSL)](#test-specification-language).
+<a name="application-mapping"></a>Application Mapping
+: A set of platform- or application-specific [test steps](#test-step), mapped to concrete actions to be performed for testing, that are implemented in [test fixtures](#test-fixture).
+
+<a name="application-mapping-language"></a>Application Mapping Language (AML)
+: A language used by [testers](#tester) to define platform- or application-specific [test steps](#test-step), and by [developers](#developers) to map these steps to the concrete actions to be performed for testing, implemented in [test fixtures](#test-fixture).
```
Comment (gunther-bachmann) [inline te_markdown/terminology.md]:
```suggestion
: A language used by [testers](#tester) to define the platform- or application-specific vocabulary for the [test steps](#test-step), and by [developers](#developers) to map these steps to the concrete actions to be performed for testing, implemented in [test fixtures](#test-fixture).
```

File: te_markdown/terminology.md [hunk 2]
```diff
@@ -35,25 +35,47 @@ In this area, details about the test step selected in the _Test Execution Naviga
...
+
+<a name="domain-expert"></a>Domain Expert
+: A user of the [software under test](#software-under-test), or other stakeholder, who has a deep understanding of the problem domain. Domain experts specify what they expect the system to do and how it should behave, using the [test specification language (TSL)](#test-specification-language).
 
 <a name="developer"></a>Developer
-: TBW
+: A person who is involved in the development of the [software under test](#software-under-test). Developers execute user acceptance tests described with the [test case language (TCL)](#test-case-language) to check their progress with each development increment, and to validate their implementation.
```
Comment (gunther-bachmann) [inline te_markdown/terminology.md]:
```suggestion
: A person who is involved in the development of the [software under test](#software-under-test). User acceptance tests described with the [test case language (TCL)](#test-case-language) are continuously executed during the development process to check the progress with each development increment, and to validate the implementation.
```

File: te_markdown/terminology.md [hunk 3]
```diff
@@ -35,25 +35,47 @@ In this area, details about the test step selected in the _Test Execution Naviga
...
+
+<a name="test-editor-web"></a>Test-Editor-Web
+: An application dedicated to automate the tip of the testing pyramid—acceptance testing—to enable automated product releases and provide confidence in reliable product operation. Designed to bridge the gap between [domain experts](#domain-expert), [testers](#tester), and [developers](#developer), Test-Editor-Web is a modern application that integrates well with container technologies. See (Overview)[#overview] for a description of its user interface components and functions.
+
+<a name="test-fixture"></a>Test Fixture
+: A software library that implements concrete actions, specific to the platform and technology stack of a given [software under test](#software-under-test), to be performed during test execution. [Developers](#developer) use [application mappings](#application-mapping) to bind [test-steps](#test-step) to test fixture logic. Out of the box, [Test-Editor-Web](#test-editor-web) comes with fixtures that enable the testing of _web_ applications, in particular _Angular_ applications, _REST_ services, _Java Swing_ applications, and mainframe applications through _3270 terminal emulation_.
```
Comment (gunther-bachmann) [inline te_markdown/terminology.md]:
```suggestion
: A software library that implements concrete actions, specific to the platform and technology stack of a given [software under test](#software-under-test), to be performed during test execution. [Developers](#developer) use [application mappings](#application-mapping) to bind [test-steps](#test-step) to fixture logic. Out of the box, [Test-Editor-Web](#test-editor-web) comes with fixtures that enable the testing of _web_ applications, in particular _Angular_ applications, _REST_ services, _Java Swing_ applications, and mainframe applications through _3270 terminal emulation_.
```

File: te_markdown/terminology.md [hunk 4]
```diff
@@ -35,25 +35,47 @@ In this area, details about the test step selected in the _Test Execution Naviga
...
+<a name="test-specification"></a>Test Specification
+: A usage scenario describing an interaction with the [software under test](#software-under-test), written by [domain experts](#domain-expert) to express desired system behaviors in a testable manner.
 
 <a name="test-specification-language"></a>Test Specification Language (TSL)
-: TBW
--->
+: A language used by [domain experts](#domain-expert) to write [test specifications](#test-specification) of desired system behaviors. [Testers](#tester) refer to TSL [artifacts](test-artifact) when implementing them as [test cases](#test-case), using the [test case language](#test-case-language).
+
+<a name="test-step"></a>Test Step
+: A single, automatable step taken as part of a test executed against the [software under test](#software-under-test). In [test cases](#test-case) conforming to [test specifications](#test-specification), each [specification step](#specification-step) may be implemented by arbitrarily many test steps.
```
Comment (gunther-bachmann) [inline te_markdown/terminology.md:78]:
```suggestion
: A single, automatable step taken as part of a test executed against the [software under test](#software-under-test). In [test cases](#test-case) conforming to [test specifications](#test-specification), each [specification step](#specification-step) may be implemented by arbitray many test steps.
```

Comment (jjelschen) [reply te_markdown/terminology.md:78]: I assume you meant to write "arbitrary", but I think we should use an adverb here (even though "many" isn't really a verb – but its also not a noun, obviously, so I don't know…).

**Pull request merged**